cogs/modmail.py

```diff
@@ -1568,6 +1639,25 @@ async def edit(self, ctx, message_id: Optional[int] = None, *, message: str):
     @checks.has_permissions(PermissionLevel.REGULAR)
     async def selfcontact(self, ctx):
         """Creates a thread with yourself"""
+        # Check if user already has a thread
+        existing_thread = await self.bot.threads.find(recipient=ctx.author)
+        if existing_thread:
+            if existing_thread.snoozed:
+                # Unsnooze the thread
+                await ctx.send(f"ℹ️ You had a snoozed thread. Unsnoozing now...")
```

Comment:
Possibly add a delete_after here so this info message is removed and doesn't linger. Alternatively store the message object and delete after the unsnooze is finished